src/components/TransactionForms/CustomForms/Deposit.tsx

```diff
@@ -19,12 +26,75 @@ const Deposit: React.FC<PropsWithChildren<IContractProps>> = ({
   handleFormSubmit,
 }) => {
   const { handleSubmit, watch } = useFormContext<FormData>();
+  const [currencyValue, setCurrencyValue] = useState<string>('');
+  const [currencyIsValid, setCurrencyIsValid] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+
   const depositType: number = watch('depositType');
 
   const onSubmit = async (data: FormData) => {
     await handleFormSubmit(data);
   };
 
+  useEffect(() => {
+    const validateCurrency = async (): Promise<void> => {
+      if (currencyValue === '') {
+        setCurrencyIsValid(false);
+        setError('Invalid Currency');
+        return;
+      }
+
+      setIsLoading(true);
+      setError('');
+
+      try {
+        const fetchResponse = fetch(
+          `${
+            process.env.DEFAULT_API_HOST || 'https://api.testnet.klever.finance'
```

Comment:
there is already an asset request in the project, check at src/services/requests/asset ou assets

Reply:
Done!